### PR TITLE
Feat: Carrousel logic hook

### PR DESCRIPTION
**Pull request by italomlp**
Add hooks for:
- window dimensions
- media queries
- carrousel logic

### CODE REVIEW

File: hooks/useTsCarrousel.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  /**
+   * Determines if the carrousel should cycle through items automatically.
+   */
+  shouldCycle?: boolean;
+  /**
+   * The delay in milliseconds for auto-changing items in the carrousel. If greater than 0,
+   * this overrides `shouldCycle` property to true.
+   */
+  autoChangeDelay?: number;
+};
```
Comment (sgtbrunner) [inline hooks/useTsCarrousel.tsx:32]: Noice documentation!

File: hooks/useTsCarrousel.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+export function useTsCarrousel<T>({
+  items,
+  visibleItemsCountParam,
+  shouldCycle: shouldCycleParam = false,
+  autoChangeDelay = 0,
+}: Params<T>) {
+  const hoverItem = useSignal<T | undefined>(undefined);
+  const currentIndex = useSignal(0);
+  const { currentMediaQuery } = useTsMediaQuery();
+  console.log("currentMediaQuery", currentMediaQuery);
```
Comment (sgtbrunner) [inline hooks/useTsCarrousel.tsx:43]: Do we need this log here?

File: hooks/useTsCarrousel.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  const shouldCycle = shouldCycleParam || autoChangeDelay > 0;
+
+  const visibleItemsCount = useMemo(() => {
+    console.log(
+      "currentMediaQuery",
+      currentMediaQuery,
```
Comment (sgtbrunner) [inline hooks/useTsCarrousel.tsx:49]: Do we need this log here?

File: hooks/useTsCarrousel.tsx (hunk 1 above)
Comment (italomlp) [reply hooks/useTsCarrousel.tsx:32]: thanks to Codeium hahaha

File: hooks/useTsCarrousel.tsx (hunk 3 above)
Comment (italomlp) [reply hooks/useTsCarrousel.tsx:49]: oh, thanks, I missed it

**Pull request merged**